2-projects/1-todo-list/script.js

```diff
@@ -1,25 +1,101 @@
-function populateTodoList(todos) {
-  let list = document.getElementById("todo-list");
-  // Write your code to create todo list elements with completed and delete buttons here, all todos should display inside the "todo-list" element.
+// function populateTodoList(todos) {
+// let list = document.getElementById("todo-list");
+// Write your code to create todo list elements with completed and delete buttons here, all todos should display inside the "todo-list" element.
+
+function toggleIsComplete(todo) {
+  todo.completed = !todo.completed;
+}
+
+function greenRawTodo(li, isComplete) {
+  if (isComplete) {
+    li.style.textDecoration = "line-through";
+    li.style.color = "red";
+  } else {
+    li.style.textDecoration = null;
+    li.style.color = "black";
+  }
 }
 
+function deleteTodo(li) {
+  document.querySelector("ul").removeChild(li);
+}
 // These are the same todos that currently display in the HTML
 // You will want to remove the ones in the current HTML after you have created them using JavaScript
 let todos = [
   { task: "Wash the dishes", completed: false },
   { task: "Do the shopping", completed: false },
 ];
+// Add an event listener to the delete button
+function PopulateTodoList(todo) {
+  const li = document.getElementById("todo-list");
+  todo.forEach((todo) => {
+    const li = createDOMListItem(todo);
+    list.appendChild(li);
+  });
+}
+
+function createDOMListItem(todo) {
+  const li = document.createElement("li");
+  const tickIcon = document.createElement("i");
+  tickIcon.classList.add("fa", "fa-check");
+
+  const binIcon = document.createElement("i");
+  binIcon.classList.add("fa", "fa-trash");
 
-populateTodoList(todos);
+  if (!tickIcon.onclick) {
+    tickIcon.addEventListener("click", () => {
+      toggleIsComplete(todo);
+      greenRawTodo(li, todo.completed);
+    });
+  }
+  if (!binIcon.onclick) {
+    binIcon.addEventListener("click", () => deleteTodo(li));
+  }
 
+  if (todo.completed) {
+    li.style.textDecoration = "line-through";
+  }
+
+  if (todo.task.length < 1) {
```

Comment:
this is fine, but it implies that `.length` can be negative which could be confusing for other deveopers. In this case, you'd be better specifying `.length === 0` as it's the fact it's 0 you care about, not that it's specifically less than 1.